Updates

gcomoretto opened the pull request:
Few improvements plus NCSA inputs.

gcomoretto committed:
JSwinbank comments

gcomoretto committed:
Added other contributors

gcomoretto committed:
Added NCSA comments

gcomoretto committed:
Comments from NCSA

gcomoretto committed:
Added generalization

gcomoretto committed:
Adde MButler and MGelman as coauthors

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
 \section{Introduction} \label{sec:intro}
 
-The current situation is  that, only the Science Pipelines is released as follows:
+This document collects the needs in therms of releses from different stakehoders, section \ref{sec:reqs}.
```
This document presents the requirements for \gls{LSST} software releases from the various stakeholders. These requirements, detailed in section \ref{sec:reqs}, are not formal project requirements,  as given for example in the \gls{DMS} requirements specification \citeds{LSE-61}.

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
 
-The current situation is  that, only the Science Pipelines is released as follows:
+This document collects the needs in therms of releses from different stakehoders, section \ref{sec:reqs}.
+These needs are presented as requirements, that shall not be confused with project requirements, like for example the \gls{DMS} requirement specification, documented in \citeds{LSE-61}.
```
This document presents the requirements for \gls{LSST} software releases from the various stakeholders. These requirements, detailed in section \ref{sec:reqs}, are not formal project requirements, as given for example in the \gls{DMS} requirements specification \citeds{LSE-61}.

jdswinbank commented:
Hi Gabriele — I know we discussed this last week, but I'm afraid I can't remember what we agreed about it, and it still seems wrong to me in this draft —

Much of the text in the “requirements” section is not about requirements. For example, it is not a requirement that “back-porting bug fixes to the current stable version of the software will entail extra work” (that's a simple statement of fact), and nor are particular actions which the DM-SST might take to liaise with the community requirements.

I think the document would be better served by using the requirements section _just_ to list requirements, and moving all this sort of commentary into the “assessment” section.

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
+Section \ref{sec:process} provides an overview of the release process.
+
+Appedix \ref{sec:defs} provides definitions of the main objects involved in the release activities. 
+Those definitions are given in order to provide a common understanding.
```
These definitions provide a common understanding of the components.

gcomoretto replied:
OK, I would not use the word "components". I better suggest "elements involved"

leannep replied:
Just say: 
Appendix \ref{sec:defs} provides definitions of the main objects involved in the release activities in order to provide a common understanding.

I don't like introducing a new term

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
+
+Appedix \ref{sec:defs} provides definitions of the main objects involved in the release activities. 
+Those definitions are given in order to provide a common understanding.
+In the same line, appendix \ref{sec:swptree} gives an overview of the software products part of the \gls{DM}, that are involved in this process.
```
appendix \ref{sec:swptree} also provides an overview of the software products part of the \gls{DM} system that are involved in this process.

gcomoretto replied:
I suppose the verb needs to be kept...

leannep replied:
Yes- I just updated the original comment

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
+\section{Science Pipelines Releases}\label{sec:sci}
+
+The current situation is that only the Science Pipelines product is released regularly.
+The process in place provides time based builds and releases:
```
Currently, only the Science Pipelines product is released. Builds and releases are made on a the following time-based cadence:

leannep replied:
Builds and releases are made on a the following time-based cadence: -> remove `a`

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
+Technical details and problems are given in \citeds{DMTN-106}.
 
-For this reasons weekly builds are largely used, since they provide the most up to date and stable software.
+Due to the time issue, when a release is available to the users, it is already too old.
```
Consequently,  by the time a release becomes available to the users, it is already old.

(remove the paragraph break)

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
 
-For this reasons weekly builds are largely used, since they provide the most up to date and stable software.
+Due to the time issue, when a release is available to the users, it is already too old.
+For this reason, users prefer in general weekly builds, that are sufficiently stable and include all new functionalities completed in the last week.
```
For this reason, users generally prefer to work with weekly builds that are sufficiently stable and include all new functionalities completed in the last week.

leannep commented on process.tex:
```diff
@@ -1,5 +1,10 @@
 
+
+This high level process has already been documented in \citeds{LDM-294}, section 3.6. 
+In order to avoid duplications, this process should be maintained only in one document.
```
The high-level Release Management plan for  \gls{DM} is documented in \citeds{LDM-294},

gcomoretto replied:
A decision has to be taken: if this section has to be maintained here, or in LDM-294 and just refer to it.
I am more inclined for the second option, remove this section from here and add in the introduction a reference to LDM-294, section 3.6.

leannep replied:
To me, the statement 'The high-level Release Management plan for \gls{DM} is documented in \citeds{LDM-294},' implies that LDM-294 is the main source.

leannep commented on requirements.tex:
```diff
@@ -3,7 +3,12 @@ \section{Requirements} \label{sec:reqs}
 In this section we give an overview of their respective requirements on  \gls{DM} software product releases.
 
-To be kept in mind that the content of a release, must be well understood and documented in the corresponding release note.
+It is important to identify the correct release policy during contraction phase, in order to have it consolidated when operations starts.
```
In this section we give an overview of requirements on \gls{DM} software product releases from the different stakeholders.

leannep commented on requirements.tex:
```diff
@@ -3,7 +3,12 @@ \section{Requirements} \label{sec:reqs}
+It is important to identify the correct release policy during contraction phase, in order to have it consolidated when operations starts.
+
+Once in operations, some of the LSST subsystems mentioned here may not be relevant anymore, but the policy defined here shall still be applicable, since many of the stakeholders will still expect software releases to be managed properly. 
+
```
Move this to the subsection on operations

gcomoretto replied:
This is paragraph is not just for operational (processing) software but is applicable in general. Infrastructural or other subsystems SW will still be used (and maintained) during operations even if DM (or T&S) are not LSST operational subsystems.

leannep commented on requirements.tex:
```diff
@@ -3,7 +3,12 @@ \section{Requirements} \label{sec:reqs}
 There are several different stakeholders of the \gls{LSST} \gls{DM} software products, each of whom may have different requirements for releases.
```
remove

leannep commented on requirements.tex:
```diff
@@ -3,7 +3,12 @@ \section{Requirements} \label{sec:reqs}
 There are several different stakeholders of the \gls{LSST} \gls{DM} software products, each of whom may have different requirements for releases.
 In this section we give an overview of their respective requirements on  \gls{DM} software product releases.
```
remove

leannep commented on requirements.tex:
```diff
@@ -24,10 +29,17 @@ \subsection{Release Requirements for the \gls{LSST} Science Community} \label{se
-The Science Operations Department will be in charge of  data processing in operations. 
-Software for the various pipelines will be hosted at \gls{NCSA},IN2P3 and in Chile.
+%The Science Operations Department will be in charge of  data processing in operations. 
+%Software for the various pipelines will be hosted at \gls{NCSA},IN2P3 and in Chile.
```
In operations, some of the LSST subsystems may no longer exist, for example \gls{DM} will not exist per se in operations but \gls{DM} products, such as the Science Pipelines, will still exist under some other managerial structure. 
The policy defined here will still be applicable, since many of the stakeholders will still expect software releases to be managed in the same manner as during construction.

gcomoretto replied:
Fine. With the caveat as I mentioned before, this is not just Operational Processing. I also think it is not good to mention Science Pipelines.

leannep commented on requirements.tex:
```diff
@@ -24,10 +29,17 @@ \subsection{Release Requirements for the \gls{LSST} Science Community} \label{se
+%The Science Operations Department will be in charge of  data processing in operations. 
+%Software for the various pipelines will be hosted at \gls{NCSA},IN2P3 and in Chile.
+
+The \gls{LDF} will be responsible for generating the needed data products during commissioning and operations. 
```
The \gls{LDF} will be responsible for generating the necessary data products during commissioning and operations.

leannep commented on requirements.tex:
```diff
@@ -24,10 +29,17 @@ \subsection{Release Requirements for the \gls{LSST} Science Community} \label{se
+
+The \gls{LDF} will be responsible for generating the needed data products during commissioning and operations. 
+The \gls{LDF} requires officially released software to be used in production for the various operational activities.
+Software releases shall be properly documented with a software release note as per definition in \ref{sec:defrelease} and tested before making them available for operations.
```
This is a general requirement applicable to all stakeholders, i.e Software releases must *always* be documented and tested before release. Move this comment to a higher level so that it is applicable to all stakeholders.

gcomoretto replied:
Right, I move this in a specific subsection.

leannep commented on requirements.tex:
```diff
@@ -24,10 +29,17 @@ \subsection{Release Requirements for the \gls{LSST} Science Community} \label{se
+The \gls{LDF} requires officially released software to be used in production for the various operational activities.
+Software releases shall be properly documented with a software release note as per definition in \ref{sec:defrelease} and tested before making them available for operations.
+Software releases will be run in production at National Center for Supercomputing Applications (\gls{NCSA}), CC-IN2P3 and in Chile. 
+Additional processing centers may be added to the list.
```
Software releases will be run in production at National Center for Supercomputing Applications (\gls{NCSA}), CC-IN2P3 and in Chile, and possibly at independent Data Access Centers (iDACs)

leannep commented on requirements.tex:
```diff
@@ -64,3 +78,10 @@ \subsection{Release Requirements for non-Operational Project Activities} \label{
+
+\subsection{Assessment} \label{sec:assessment}
+
+As a first glance, it seems  that major releases need to be done more on a feature or milestone bases. 
```
As a first glance, it seems that major releases need to be done more on a feature or milestone basis.

leannep commented on requirements.tex:
```diff
@@ -64,3 +78,10 @@ \subsection{Release Requirements for non-Operational Project Activities} \label{
+As a first glance, it seems  that major releases need to be done more on a feature or milestone bases. 
+
+However before this assessment need further evaluation and clearly identify which of the above requirements is applicable for each of the software products  in appendix \ref{sec:swptree}.
+This extra information shall be included in MagicDraw and documented in \citeds{DMTN-104}, still draft.
```
However before concluding, further evaluation is needed to clearly identify which of the above requirements is applicable for each software product listed  in appendix \ref{sec:swptree}.
This extra information shall be included in MagicDraw and documented in \citeds{DMTN-104}, which still draft.

leannep reviewed (changes requested):
Please address the comments

gcomoretto commented:
> Hi Gabriele — I know we discussed this last week, but I'm afraid I can't remember what we agreed about it, and it still seems wrong to me in this draft —
> 
> Much of the text in the “requirements” section is not about requirements. For example, it is not a requirement that “back-porting bug fixes to the current stable version of the software will entail extra work” (that's a simple statement of fact), and nor are particular actions which the DM-SST might take to liaise with the community requirements.
> 
> I think the document would be better served by using the requirements section _just_ to list requirements, and moving all this sort of commentary into the “assessment” section.

Hi John, I understood we agreed to clarify that those are not requirements in the sense of project requirements. 

Each statement can be rephrased with a "shall", in order to make them more formal. 
A bullet list of requirements can be given in the assessment subsection, after assessing the different inputs.

gcomoretto committed:
Improvements

gcomoretto commented:
I have pushed a commit that resolves all suggestions from @leannep. I have also reviewed the Introduction and added some more content in the assessment subsection, trying to address @jdswinbank comment.

jdswinbank commented:
Hi @gcomoretto — sorry if my above comment was unclear. I think there are two issues here which are getting confused.

1. Many of the things which are listed as “requirements” in this document are not requirements in the formal sense of the SRD, LSR, OSS, DMSR, etc. They are requests or desires from stakeholders, the community, and so on, which do _not_ bind the project in the same way as the formal requirements documentation. When we discussed this previously, you suggested adding a qualification in the introduction. You have now done this (“These requirements... are not formal project requirements”). That's fine.
2. Many of these aren't even desires or requests from stakeholders; rather, they are commentary on the above. For example, take the statement “back-porting bug fixes to the current stable version of the software will entail extra work for \gls{DM}”. No stakeholder requests or desires that back-porting fixes should take DM extra work — all they care about is that the back-porting takes place. The statement about extra work is purely commentary on these “requirements”; it is not a requirement itself, and should be removed from the requirements section.

gcomoretto committed:
removed appendix b

gcomoretto commented:
> 2\. Many of these aren't even desires or requests from stakeholders; rather, they are commentary on the above. For example, take the statement “back-porting bug fixes to the current stable version of the software will entail extra work for \gls{DM}”. No stakeholder requests or desires that back-porting fixes should take DM extra work — all they care about is that the back-porting takes place. The statement about extra work is purely commentary on these “requirements”; it is not a requirement itself, and should be removed from the requirements section

Agreed, that sentence seems not to be relevant from a release policy point of view, and in any case, it shall be addressed in the Process extracted from LDM-294. @leannep what do you think?

gcomoretto committed:
Removed out of scope paragraphs

gcomoretto commented:
I have removed (just commented out for the moment) the two paragraphs and added an extra bullet to the bullet list.

@leannep can you please approve the PR, so I can merge it to master and RFC the document?

jdswinbank committed:
Need \makeglossaries before defining entries.

If those entries cross-reference each other. Per manual, §8.

jdswinbank committed:
Ignore .old files.

jdswinbank committed:
Fix build by referring to lsst-texmf branch.

jdswinbank committed:
Merge branch 'u/swinbank/DM-19698-fix-build' into tickets/DM-19698

gcomoretto commented:
@leannep , from our conversations last week it seems that the PR is ready to go.
Formally, it requires still your approval after my last changes.

gcomoretto committed:
Update README

leannep commented on LDM-672.tex:
```diff
@@ -39,8 +39,9 @@
+  \addtohist{}{2019-05-09}{Clarifications}{G.~Comoretto}
   \addtohist{}{2019-05-07}{Updated community requirements, abstractt}{L.~Guy}
-  \addtohist{}{2019-04-22}{Definitions and Requirements}{G.~ Comoretto}
+  \addtohist{}{2019-04-22}{Definitions and Requirements}{G.~Comoretto}
```
Add in change record to include contributions from K-T, Margaret, Michelle (and others if any)

leannep commented on introduction.tex:
```diff
@@ -1,15 +1,31 @@
+These needs are presented as requirements, that shall not be confused with project requirements, like for example the \gls{DMS} requirement specification, documented in \citeds{LSE-61}.
+Section \ref{sec:process} provides an overview of the release process.
+
+Appedix \ref{sec:defs} provides definitions of the main objects involved in the release activities. 
```
Appendix

gcomoretto committed:
Updated changelog

gcomoretto committed:
Moveed definitions to DMTN016, updated changelog

gcomoretto committed:
Removed process section and rephrased

gcomoretto commented:
After talking to @jdswinbank, I have also removed the `Process` section, and added a couple of subsection in the introduction, one for the Release Management Process referring to LDM-294 and one for the definitions, referring to DMTN-106.
I have also rephrased accordingly some text in the requirements section.

gcomoretto committed:
Added meta.tex

gcomoretto committed:
General requirements rephrase

Commit:
Corrected names, curator

leannep reviewed (approved):
ready for RFC

gcomoretto committed:
Glossary cleanup

gcomoretto committed:
Removed unnecessary files

gcomoretto committed:
Fix

gcomoretto committed:
Remove Asessment and draft ref to 106

gcomoretto committed:
Added policy and 3rd party packages reqs

womullan committed:
tidy 3rd party

gcomoretto committed:
Review following Leanne's comments, added sect 6

womullan committed:
fixes

gcomoretto committed:
Merge branch 'u/womullan/fixes' into tickets/DM-19698

gcomoretto committed:
Additional WOM comments

gcomoretto committed:
Corrections

gcomoretto committed:
Correct citation to DevGuide

gcomoretto committed:
Small changes agreed with Leanne

jdswinbank committed:
Fix TEXMFHOME.

jdswinbank committed:
Remove duplicate material.

jdswinbank committed:
Minor rewording of requirements section.

jdswinbank committed:
Tweaks to consolidated requiremnents.

jdswinbank committed:
Tweak policy wording.

jdswinbank committed:
Reword applicability.

jdswinbank committed:
Polish release policy.

jdswinbank committed:
Fix typo.

gcomoretto committed:
Rephrase

gcomoretto committed:
Added small rephrase

gcomoretto committed:
Added backport term

jdswinbank committed:
Clean up requirements table.

jdswinbank committed:
Add me as author.

jdswinbank committed:
Merge branch 'u/swinbank/DM-19698' into tickets/DM-19698
